cs/battle.go

```diff
@@ -983,8 +999,10 @@ func (b *battle) getBestAttackMove(token *battleToken) BattleVector {
 	bestDamageRatio := 0.0
 	bestNetDamage := -math.MaxInt
 	target := token.moveTarget
-	var bestMove = token.Position
-	var lowestDamageMove = token.Position
+	var bestMoves []BattleVector
+	bestMoves = append(bestMoves, token.Position)
+	var lowestDamageMoves []BattleVector
```

Comment:
Do we know the size of these slices ahead of time? Or at least a best guess? It's more performant to pre-allocate the slice with `make`

This will make a slice of BattleVectors with initial capacity of 9, but 0 entries in it. This way when you `append` it doesn't have to reallocate anything, It just uses some existing capacity.
```golang
bestMoves := make([]BattleVector, 0, 9)
```

Reply:
I guessed 9 because the bestMoves can at most be one move in each direction, right?